ui/dialogs/searchdialog.go

```diff
@@ -0,0 +1,279 @@
+package dialogs
+
+import (
+	"image"
+	"log"
+	"sync"
+
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/container"
+	"fyne.io/fyne/v2/layout"
+	"fyne.io/fyne/v2/theme"
+
+	"fyne.io/fyne/v2/widget"
+	"github.com/dweymouth/supersonic/backend/mediaprovider"
+
+	"github.com/dweymouth/supersonic/ui/layouts"
+	myTheme "github.com/dweymouth/supersonic/ui/theme"
+	"github.com/dweymouth/supersonic/ui/util"
+	"github.com/dweymouth/supersonic/ui/widgets"
+)
+
+type SearchDialog struct {
```

Comment:
Can we add a comment here indicating that this is a base widget to be built on top of, rather than a complete dialog itself? (Could be done in a follow-up if you just want me to merge this now, I'm OK with either)

Reply:
Sure, I'll add that comment in this PR. Thanks for the suggestion!